stormpiper/stormpiper/spa/src/components/scenario-module/scenario-detail-page.jsx

```diff
@@ -62,18 +73,54 @@ export default function ScenarioDetailPage({ setDrawerButtonList }) {
 
   useEffect(() => {
     getDataByID(params.id).then((res) => {
-      setScenarioObject(res);
-      if (res.input.delineation_collection) {
-        console.log("Found delineation");
-        setDelineation(res.input.delineation_collection);
-      }
-      if (res.input.tmnt_facility_collection) {
-        console.log("Found facility: ", res.input.tmnt_facility_collection);
-        setFacility(res.input.tmnt_facility_collection);
-      }
+      buildScenario(res);
+      setCachedScenario(res);
+      setLastCalculatedAt(dateFormatter(res?.result_time_updated));
     });
   }, [params.id, resultsSuccessDisplay]);
 
+  const dateFormatter = (dtValue) => {
```

Comment:
move this to utils, we'll need it for the BMp detail page as well

Reply:
moved to utils